Engine/DataFeeds/SubscriptionCollection.cs

```diff
@@ -117,17 +126,70 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
 
+        /// <summary>
+        /// Helper method to validate a configuration to be included in the fill forward calculation
+        /// </summary>
+        private static bool ValidateFillForwardResolution(SubscriptionDataConfig configuration)
+        {
+            return !configuration.IsInternalFeed && configuration.Resolution != Resolution.Tick;
+        }
+        /// <summary>
+        /// Gets and updates the fill forward resolution by checking specified subscription configurations and
+        /// selecting the smallest resoluton not equal to tick
+        /// </summary>
+        private void UpdateFillForwardResolution(bool adding, SubscriptionDataConfig configuration)
+        {
+            // Due to performance implications let's be jealous in updating the _fillForwardResolution
+            if (ValidateFillForwardResolution(configuration) &&
+                (
+                    (adding  // we are adding
+                    && configuration.Resolution.ToTimeSpan() != _fillForwardResolution.Value) // check if the new timeSpan is different
+                ||
+                    (!adding // We are removing
+                    && configuration.Resolution.ToTimeSpan() == _fillForwardResolution.Value // True: We are removing the resolution we were using
+                    && _subscriptions.All(x => x.Key.Resolution != configuration.Resolution))) // False: there is at least another one equal, no need to update
+                )
+            {
+                var configurations = adding ? _subscriptions.Keys.Concat(new[] { configuration }) : _subscriptions.Keys;
```

Comment:
I think the name `adding` here is misleading. I think we only need to append the requested `configuration` is via `UpdateAndGetFillForwardResolution` where the consumer is basically saying, I'm 'about' the add this to the collection, so start using this one now, but via calls from `TryAdd` above, the specified `configuration` will already be in `_subscriptions.Keys` (methinks)

Reply:
yes your right! I'll make it more specific and clear